src/internal_events/topology.rs

```diff
@@ -36,3 +37,18 @@ impl InternalEvent for EventOut {
         }
     }
 }
+
+#[derive(Debug)]
+pub struct InternalReloadFailed {
+    pub error: Error,
+}
+
+impl InternalEvent for InternalReloadFailed {
+    fn emit_logs(&self) {
+        warn!(message = "Failed reloading component internal data.", error = %self.error);
```

Comment:
That should probably be an error instead of warning. This is pretty serious, I would think. Also, rate limit it. Since this can be thrown from a deref operation, it could happen very quickly.